iocore/net/P_SSLCertLookup.h

```diff
@@ -52,30 +87,42 @@ struct ssl_ticket_key_block {
 
 */
 struct SSLCertContext {
-  /** Special things to do instead of use a context.
-      In general an option will be associated with a @c nullptr context because
-      the context is not used.
-  */
-  enum Option {
-    OPT_NONE,  ///< Nothing special. Implies valid context.
-    OPT_TUNNEL ///< Just tunnel, don't terminate.
-  };
-
-  SSLCertContext() {}
-  explicit SSLCertContext(SSL_CTX *c) : ctx(c) {}
-  SSLCertContext(SSL_CTX *c, Option o) : ctx(c), opt(o) {}
-  SSLCertContext(SSL_CTX *c, Option o, ssl_ticket_key_block *kb) : ctx(c), opt(o), keyblock(kb) {}
+private:
+  mutable std::mutex ctx_mutex;
```

Comment:
Just comment, I haven't noticed we start using `std::mutex` instead of `ink_mutex`. I'd like to choose one for the consistency.

/cc @SolidWallOfCode